### PR TITLE
Refactored some code

### PR DESCRIPTION
**Pull request by Webd01**
Hey,

- I refactored some code (removed unused imports, variables, etc..).

- I added the [Discord Context](https://www.npmjs.com/package/discord-context) dependency for a cleaner code.

- An error occurred when we run the command **!addresource** without a #resources channel. So I handled that, it now sending a message that said *Please add a channel #resources*.

- I added the **config.json** file to the gitignore and created a **config.example.json** file, so we can be sure to don't leak the token accidentally.

Don't hesitate to comment if you want me to change some things. :)

### CODE REVIEW

**Comment by harrego** [comment]
@dhilln dang it ;) I wish you had merged my PR quicker, my PR (#2) makes a lot of structural changes to the project that heavily conflicts with this PR. We may need to merge mine (#2), resolve conflicts with this PR and then merge this one.

My apologies about this!

File: config.example.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (harrego) [inline config.example.json]: I don't believe replacing `config.json` with an example config leads to the best experience for the user. Quick start is important to a bot, setting the key and then running is essential in my opinion.

File: config.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "prefix": "!",
-  "token": "",
+  "token": "NDg0MTI5Nzk5NDgxMTk2NTUx.Dm4Nlw.SclBOhr81x67EU1jYi6iLH19JYI",
```
Comment (harrego) [inline config.json]: In future please make sure to clean your git commits if you leak the token!

File: handlers/message.js [hunk 3]
```diff
@@ -1,19 +1,16 @@
+const DiscordContext = require('discord-context')
+
 // Fired when a message is sent
 function handle(client, message, args) {
+  const context = new DiscordContext(message)
```
Comment (harrego) [inline handlers/message.js]: I'm excited to have this in the project! Looks like a nice module!

File: config.json (hunk 2 above)
Comment (Webd01) [reply config.json]: Oh god, I completely forgot this. Sorry

**Comment by harrego** [comment]
@Webd01 #2 is now merged, if you could resolve the conflicts then I'll give this PR another review! Thanks

**Comment by Webd01** [comment]
@harrego All conflicts are resolved now.

**Comment by harrego** [comment]
I've looked over the code and I've decided that Discord Context perhaps isn't the best fit for the project right now as it causes an abstraction over `discord.js` which is heavily documented and works well for us.

However! I moved this commit to branch `Webd01-webd-patch-1` so I can remove the Discord Context and keep in the great factoring you contributed! I hope this isn't a problem, once again thank you for the PR!

**Comment by Webd01** [comment]
Okay, no problem.

**Pull request closed without merging**